src/lib/server/nvim-sync/config/syncRepoInfo.ts

```diff
@@ -8,89 +8,108 @@ import type { NeovimConfig, User } from '@prisma/client';
+  const repo = await fetchGithubRepositoryByName(token, owner, repoName);
+  const upsertDTO = upsertNeovimConfigDTOFactory(owner, root, init, repo);
+  const configUser = user.username === owner ? user : await getUserByUsername(owner);
+  return await upsertNeovimConfig(configUser.id, upsertDTO);
 }
 
 export async function syncReadme(token: string, config: NeovimConfig) {
-	const readme = await fetchReadme(token, config.owner, config.repo).catch(() => {
-		console.log('Error fetching readme');
-		return '';
-	});
+  const readme = await fetchReadme(token, config.owner, config.repo).catch(() => {
+    console.log('Error fetching readme');
+    return '';
+  });
 
-	const shields = ['leaderkey', 'plugins', 'plugin-manager'].map((shield) => {
-		return `https://dotfyle.com/${config.owner}/${config.slug}/badges/${shield}`;
-	});
+  const shields = ['leaderkey', 'plugins', 'plugin-manager'].map((shield) => {
+    return `https://dotfyle.com/${config.owner}/${config.slug}/badges/${shield}`;
+  });
 
-	const hasShield = shields.filter((s) => readme.includes(s)).length > 0;
+  const hasShield = shields.filter((s) => readme.includes(s)).length > 0;
 
-	if (!config.dotfyleShieldAddedAt && hasShield) {
-		await prismaClient.neovimConfig.update({
-			where: {
-				id: config.id
-			},
-			data: {
-				dotfyleShieldAddedAt: new Date()
-			}
-		});
-	}
+  if (!config.dotfyleShieldAddedAt && hasShield) {
+    await prismaClient.neovimConfig.update({
+      where: {
+        id: config.id
+      },
+      data: {
+        dotfyleShieldAddedAt: new Date()
+      }
+    });
+  }
 
-	if (config.dotfyleShieldAddedAt && !hasShield) {
-		await prismaClient.neovimConfig.update({
-			where: {
-				id: config.id
-			},
-			data: {
-				dotfyleShieldAddedAt: null
-			}
-		});
-	}
+  if (config.dotfyleShieldAddedAt && !hasShield) {
+    await prismaClient.neovimConfig.update({
+      where: {
+        id: config.id
+      },
+      data: {
+        dotfyleShieldAddedAt: null
+      }
+    });
+  }
 }
 
 export async function syncExistingRepoInfo(token: string, config: NeovimConfig) {
-	const repo = await fetchGithubRepositoryByName(token, config.owner, config.repo);
-	const upsertDTO = upsertNeovimConfigDTOFactory(config.owner, config.root, config.initFile, repo);
-	return upsertNeovimConfig(config.userId, upsertDTO);
+  const repo = await fetchGithubRepositoryByName(token, config.owner, config.repo);
+  const upsertDTO = upsertNeovimConfigDTOFactory(config.owner, config.root, config.initFile, repo);
+  return upsertNeovimConfig(config.userId, upsertDTO);
+}
+
+export async function deleteNeovimConfig(config: NeovimConfig) {
+  return await prismaClient.neovimConfig.delete({
+    where: {
+      id: config.id
+    }
+  });
+}
+
+export async function markForDeletion(config: NeovimConfig) {
+  return await prismaClient.neovimConfig.update({
+    where: {
+      id: config.id
+    },
+    data: {
+      markedForDeletion: true
+    }
+  });
```

Comment:
These 2 functions would be a better fit in `prisma/neovimconfigs/service.ts`

But I think they might not be necessary since you also put a function in the service file?